### DISCUSSION TITLE
[Question] Change filepath with template?

### DISCUSSION THREAD

**Discussion by Sarcas666** [khakers/go-subgen#8] [General]
I'm not very good at this stuff, but I know how to tinker a bit... I got the original Subgen docker running on my NAS, but that was way too slow. Can I make this work running the docker on my Windows PC, while everything else (Tautulli, mediafiles etc) are on the NAS?

**Comment by khakers** [comment]
Go-subgroup currently operates on the assumption that the filepath for itself and whatever sends it the notification (sonarr, radarr, tautulli, etc...) is the same. If you have your nas file system mounted on your pc or docker container where it can access everything via the exact same path then you shouldn’t have any issues.

**Reply by Sarcas666** [reply]
Ah, I have issues then…. Tautulli, all the *arr applications and the media files (/volume1/PlexMedia/) are in the Synology NAS. The Go-subgen would be on my Windows machine with a mounted share like Z:/PlexMedia. Hm, I’ll see if I can tinker myself around the problem...

**Reply by khakers** [reply]
If you're running in docker you may be able to mount the share with a path that matches your NAS (for example "/mnt/z/PlexMedia:/volume1/PlexMedia") or if that's not possible and you have NFS available NFS mount your NAS directly to your docker container

**Reply by Sarcas666** [reply]
Good idea. Currently struggling with Docker Desktop though. Keeps giving me a
`docker: Error response from daemon: Cannot mkdir: /subgen is not a directory.`
My first time using Docker Desktop, so I'm sure I messed up somewhere.

**Reply by khakers** [reply]
That may have been an error in the dockerfile on my part. I pushed a change that will hopefully fix it.

**Reply by Sarcas666** [reply]
Lol, well, I learned a lot while trying to figure out what was happening. I have to wait until tomorrow before I can try it. I’ll be back…

**Reply by Sarcas666** [reply]
Pulled the latest, but no luck. Am I doing something wrong?
`docker run -d --name=go-subgen -v go-subgen-vol:/config -v z:/:/volume1/PlexMedia ghcr.io/khakers/go-subgen:master
docker: Error response from daemon: Cannot mkdir: /subgen is not a directory.
See 'docker run --help'.`

**Reply by khakers** [reply]
From a brief looks it's possible the windows drive path might be confusing it. I think this is how windows paths are intended to be written but I haven't had time to check.
```
docker run -d --name=go-subgen -v go-subgen-vol:/config -v /z/:/volume1/PlexMedia ghcr.io/khakers/go-subgen:master

```

**Reply by Sarcas666** [reply]
`docker run -d --name=go-subgen -v go-subgen-vol:/config`
or even
`docker run -d --name=go-subgen`
give the same error though.

**Reply by khakers** [reply]
> `docker run -d --name=go-subgen -v go-subgen-vol:/config` or even `docker run -d --name=go-subgen` give the same error though.

Yeah that's an issue on my end with the dockerfile then, I can replicate it on a different machine. I cannot for the life of me figure out why I never ran into this when testing or why it builds perfectly fine when this seems like it should be a buildtime error

**Reply by khakers** [reply]
In the meantime while I figure out why the image built by CI is broken, you can build the image locally with this command `docker build https://github.com/khakers/go-subgen.git -t go-subgen` and use that instead of the ghcr one. It appears to work perfectly fine

**Reply by khakers** [reply]
Ok, I appear to have fixed the docker image build available from Github (ghcr.io/khakers/go-subgen:master).

For whatever reason the github actions runner was either not copying the binary into the file or somehow confusing the binary and an already existing directory? I assume this is some change with the software building the image between what docker desktop is shipping and what the github action is installing.

Regardless of whatever caused it, I actually tested the image and it worked on my machine™️.

**Reply by Sarcas666** [reply]
Quick reply; the new build works. Yay :) But I need to figure out how to properly point to my NAS (I've seen examples, will try them). I have no time right now, hopefully in the weekend ... I also have to read up on docker desktop, how to add/change environment vars etc. I hate it when I don't really know what I am doing :p

**Reply by Sarcas666** [reply]
Still trying to get it to work. I keep failing to tell the container to connect to the NAS somehow. I create a volume that is my NAS, but after that I haven't figured it out yet :) Baby steps, I'll get there. The solution is [here](https://www.reddit.com/r/docker/comments/lv61ra/mounting_docker_container_folder_to_nas_shared/), I think.

**Reply by Sarcas666** [reply]
Yay, it works :D Not entirely sure how I did it though, but mounting a network share finally worked.
